packages/pigeon/lib/objc_generator.dart

```diff
@@ -950,3 +970,23 @@ static id GetNullableObjectAtIndex(NSArray* array, NSInteger key) {
   }
   root.apis.forEach(writeApi);
 }
+
+/// Looks through the AST for features that aren't supported by the ObjC
+/// generator.
+List<Error> validateObjc(ObjcOptions options, Root root) {
+  final List<Error> errors = <Error>[];
+  for (final Api api in root.apis) {
+    for (final Method method in api.methods) {
+      for (final NamedType arg in method.arguments) {
+        if (_isEnum(root, arg.type) && arg.type.isNullable) {
+          // TODO(gaaclarke): Add line number.
+          errors.add(Error(
+              message:
+                  'Nullable enum types aren\'t support in ObjC arguments in method:${api.name}.${method.name} argument:(${arg.type.baseName} ${arg.name}).'));
```

Comment:
I guess this answers some of my questions above. But couldn't we just support it instead?

Reply:
I'm just cutting off scope to what I need for now.  I don't have a lot of time to open this up.

Reply:
Okay, this should be updated in the README though (https://github.com/flutter/packages/blob/main/packages/pigeon/README.md#null-safety-nnbd)

Reply:
done